l1-contracts/test/validator-selection/address_snapshots/add.t.sol

```diff
@@ -0,0 +1,40 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2025 Aztec Labs.
+pragma solidity >=0.8.27;
+
+import {AddressSnapshotsBase} from "./AddressSnapshotsBase.t.sol";
+import {
+  AddressSnapshotLib,
+  SnapshottedAddressSet
+} from "@aztec/core/libraries/staking/AddressSnapshotLib.sol";
+import {Epoch} from "@aztec/core/libraries/TimeLib.sol";
+
+contract AddressSnapshotAddTest is AddressSnapshotsBase {
+  using AddressSnapshotLib for SnapshottedAddressSet;
+
+  function test_WhenValidatorIsNotInTheSet() public {
+    // It returns true
+    // It increases the length
+    // It creates a checkpoint for the next epoch
+    // It does not change the current epoch
+
+    timeCheater.cheat__setEpochNow(1);
+    assertTrue(validatorSet.add(address(1)));
+    assertEq(validatorSet.length(), 0);
+
+    timeCheater.cheat__setEpochNow(2);
+    assertEq(validatorSet.length(), 1);
+
+    // Epoch remains frozen, so this number should not update
```

Comment:
The comment here saying should not update followed by two lines showing that it changes is kinda weird.

Reply:
Right, it meant that the query for epoch 0 shouldnt change